src/wp-includes/class-wp-block.php

```diff
@@ -191,6 +191,200 @@ public function __get( $name ) {
+	 * block with the values of the `text_custom_field` and `url_custom_field` post meta.
+	 *
+	 * @access private
+	 * @since 6.5.0
+	 *
+	 * @param string   $block_content Block content.
+	 * @param array    $block The full block, including name and attributes.
+	 */
+	private function process_block_bindings( $block_content ) {
+		$block = $this->parsed_block;
+
+		// Allowed blocks that support block bindings.
+		// TODO: Look for a mechanism to opt-in for this. Maybe adding a property to block attributes?
+		$allowed_blocks = array(
+			'core/paragraph' => array( 'content' ),
+			'core/heading'   => array( 'content' ),
+			'core/image'     => array( 'url', 'title', 'alt' ),
+			'core/button'    => array( 'url', 'text' ),
+		);
+
+		// If the block doesn't have the bindings property or isn't one of the allowed block types, return.
+		if ( ! isset( $block['attrs']['metadata']['bindings'] ) || ! isset( $allowed_blocks[ $this->name ] ) ) {
+			return $block_content;
+		}
+
+		$block_bindings_sources = get_all_registered_block_bindings_sources();
+		$modified_block_content = $block_content;
+		foreach ( $block['attrs']['metadata']['bindings'] as $binding_attribute => $binding_source ) {
+			// If the attribute is not in the list, process next attribute.
+			if ( ! in_array( $binding_attribute, $allowed_blocks[ $this->name ], true ) ) {
+				continue;
+			}
+			// If no source is provided, or that source is not registered, process next attribute.
+			if ( ! isset( $binding_source['source'] ) || ! is_string( $binding_source['source'] ) || ! isset( $block_bindings_sources[ $binding_source['source'] ] ) ) {
+				continue;
+			}
+
+			$source_callback = $block_bindings_sources[ $binding_source['source'] ]['get_value_callback'];
+			// Get the value based on the source.
+			if ( ! isset( $binding_source['args'] ) ) {
+				$source_args = array();
+			} else {
+				$source_args = $binding_source['args'];
+			}
+			$source_value = $source_callback( $source_args, $this, $binding_attribute );
+			// If the value is null, process next attribute.
+			if ( is_null( $source_value ) ) {
+				continue;
+			}
+
+			// Process the HTML based on the block and the attribute.
+			$modified_block_content = $this->replace_html( $modified_block_content, $this->name, $binding_attribute, $source_value );
+		}
+		return $modified_block_content;
+	}
+
+	/**
+	 * Depending on the block attributes, replace the HTML based on the value returned by the source.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $block_content Block content.
+	 * @param string $block_name The name of the block to process.
+	 * @param string $block_attr The attribute of the block we want to process.
+	 * @param string $source_value The value used to replace the HTML.
+	 */
+	private function replace_html( string $block_content, string $block_name, string $block_attr, string $source_value ) {
+		$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block_name );
```

Comment:
`$this->block_type` can be used here instead, as it's available on the instance.

Reply:
Done in 3134faa0510cf92fb5b686d8c9773ad0d2b8c5f5